traffic_control/clients/python/trafficops/__version__.py

```diff
@@ -14,4 +14,4 @@
 # limitations under the License.
 #
 
-__version__ = '1.1.1'
+__version__ = '1.1.2'
```

Comment:
Because if you try to do a `pip install -U` to get the changes, `pip` will see that you already have the same version and conclude there's nothing to do.

Reply:
We don't have this in PyPI though, at least not officially. If we did officially release to PyPI, I'd think we'd still follow the same version as the rest of the release. I guess if someone is installing directly from the repo's master branch the version would need to be updated with every change, but shouldn't that only be developers who should really just be using `-e` anyways?

Reply:
since it's not hosted on PyPI (officially) the only way _anyone_ could install it is through the repository - either by manually downloading the repository or using `pip`'s "install from git repo" functionality. Even a developer should be able to expect versioning. To quote what I was told about the TO API: it's versioned separately because it's a separate thing. (And unlike the TO API I totally agree that this thing should not dictate release numbers for ATC)

Reply:
TC users who use the actual TC releases probably don't really care about the version of a client, they will just use the client that comes in the release or install directly from the github using the branch that corresponds to the release (although the latter appears undocumented). Installing directly from github uses the master branch by default I think, and I'd say that should never be done for production use as it's too unstable. Developers that _need_ to use the master branch of the client should clone it, install it with `-e`, and upgrade by doing a `git pull`. I'm not completely against versioning the client like this, but it just seems like unnecessary work and will likely be overlooked most of the time. It's also not clear from the version which versions of the TO API the client actually supports.

Reply:
It's unfortunately not clear from the client code at all which version(s) of the API are supported. With our current function-decorator-based approach to adding functionality, we've made it very easy to add trivial JSON endpoints, but also each endpoint must explicitly list all of the API versions that can be used to call it. Incidentally, these `toaccess` functions bypass that by requiring the user to provide the request path and just using the highest version that existed at the time the. It's not perfect, but a fix would mean (more or less) re-writing every method of the TOSession object and probably parts of `restapi.py`.

At any rate, I'd like to keep incrementing the version number on changes, because of things that _do_ track the version number, even if that's not a common use-case. One example is the unofficial hosting on PyPi. I know that's not our responsibility, but it means that adding a changed version would require either overwriting an existing version or uploading a package that has a different `__version__` from the actual source. Since we don't have a real release process for the client, *not* incrementing it separate from ATC means that unofficial hostings of the package cannot be updated between ATC releases, which can be like a year at a time. I don't think the Python client is mature and stable enough to wait that long between adding functionality - many endpoints don't have methods, the vast majority of endpoints that _do_ have endpoints can't be called with the latest API version etc.